hack/setup-kind-in-ci.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+#
+# Refer to the --help output for a description of this script and its available options.
+#
+
+infomsg() {
+  echo "[INFO] ${1}"
+}
+
+helpmsg() {
+  cat <<HELP
+This script will run setup a KinD cluster for testing Kiali against a real environment in CI.
+Options:
+-dorp|--docker-or-podman <docker|podman>
+    What to use when building images.
+    Default: docker
+-iv|--istio-version <#.#.#>
+    The version of Istio you want to install.
+    This option is ignored if -ii is false.
+    If not specified, the latest version of Istio is installed.
+    Default: <the latest release>
+HELP
+}
+
+# process command line arguments
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -dorp|--docker-or-podman)     DORP="$2";                  shift;shift; ;;
+    -h|--help)                    helpmsg;                    exit 1       ;;
+    -iv|--istio-version)          ISTIO_VERSION="$2";         shift;shift; ;;
+    *) echo "Unknown argument: [$key]. Aborting."; helpmsg; exit 1 ;;
+  esac
+done
+
+# abort on any error
+set -e
+
+# set up some of our defaults
+DORP="${DORP:-docker}"
+
+# If a specific version of Istio hasn't been provided, try and guess the right one
+# based on the Kiali branch being tested (TARGET_BRANCH) and the compatibility matrices:
+# https://kiali.io/docs/installation/installation-guide/prerequisites/
+# https://istio.io/latest/docs/releases/supported-releases/
+if [ "${TARGET_BRANCH}" == "v1.48" ]; then
+  ISTIO_VERSION="1.13"
+elif [ "${TARGET_BRANCH}" == "v1.36" ]; then
+  ISTIO_VERSION="1.10"
+elif [ "${TARGET_BRANCH}" == "v1.24" ]; then
+  ISTIO_VERSION="1.7"
```

Comment:
I don't think these version strings work with our hack script. I think you need x.y.z. So 1.13.0 or 1.10.0 or 1.7.0. If I try just with x.y this is what I get:

```bash
$ hack/istio/download-istio.sh -iv 1.10
Output Directory: /home/jmazzite/source/kiali/kiali/_output
Will use a specific Istio version: 1.10
Will look for Istio here: /home/jmazzite/source/kiali/kiali/_output/istio-1.10
Cannot find Istio 1.10 - will download it now...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  4926  100  4926    0     0   4805      0  0:00:01  0:00:01 --:--:--  4805

Downloading istio-1.10 from https://github.com/istio/istio/releases/download/1.10/istio-1.10-linux-amd64.tar.gz ...

https://github.com/istio/istio/releases/download/1.10/istio-1.10-linux-amd64.tar.gz is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH
```